angular/src/app/services/wallet-manager.ts

```diff
@@ -797,7 +800,22 @@ export class WalletManager {
 
     // If the wallet type is restored, force an index process to restore the state.
     if (wallet.restored && runIndexIfRestored == true) {
-      this.communication.send(this.communication.createMessage('index', { force: true }, 'background'));
+      this.communication.send(this.communication.createMessage('index', {force: true}, 'background'));
+    }
+
+    if (network.smartContractSupport) {
+      await this.LoadStandardTokensForAccountAsync(network, this.getAddressByIndex(account, 0, 0));
+    }
+  }
+
+  async LoadStandardTokensForAccountAsync(network: Network, address: string) {
+    const indexerUrl = this.networkLoader.getServer(network.id, this.settings.values.server, this.settings.values.indexer);
+
+    const tokens = await axios.get(`${indexerUrl}/api/query/${network.name}/tokens/${address}`);
+    if (tokens.data.items) {
```

Comment:
What if previously you received data, but no more, shouldn't it be removed then?

Reply:
What do you mean? should I empty the store before updating it?

Reply:
If I have tokens, and I sell them, won't it still tell me that I have tokens that I no longer have? If items is null, 0, undefined, it won't be updated in the store. So if I had a token yesterday, sold it today, will it disappear or not?

Reply:
I changed the flow to clear the tokens when fetching new ones for the account